main.cpp

```diff
@@ -40,28 +40,516 @@ Purpose:  This project continues developing Project3.
  You can resolve any [-Wpadded] warnings by adding -Wno-padded to the list of compiler arguments in the .replit file. all of the "-Wno" in that file are compiler arguments.
 
  */
-
+#include <iostream>
+#include <iomanip>
+#include <vector>
 /*
  copied UDT 1:
  */
 
+struct AquariumTank
+{
+    float waterCapacity, pHLevel, currentWaterLevel, evaporationRate;
+    int fishLivingIn;
+
+    struct Fish
+    {
+        std::string name, breed;
+        float maxSalinity, age, maxAge;
+        
+        Fish();
+        ~Fish();
+
+        bool canLiveInThisWater(float waterPH, float waterSalinity);
+        void ageFish(float amountOfTime);
+        bool isStillAlive(float waterPH, float waterSalinity);
+    };
+
+    AquariumTank();
+    AquariumTank(float capacity);
+    ~AquariumTank();
+
+    void addFish(Fish fish, int quantity);
+    void addWater(float amountOfWater);
+    void adjustPH(float phAdjustment);
+
+    void ageTank(int days);
+};
+
+AquariumTank::AquariumTank() :
+waterCapacity(15.f),
+pHLevel(7.f),
+currentWaterLevel(0.8f),
+evaporationRate(0.25f),
+fishLivingIn(0)
+{
+    std::cout << "an AquariumTank was created" << std::endl; 
+}
+
+AquariumTank::AquariumTank(float capacity) :
+waterCapacity(capacity),
+pHLevel(7.f),
+currentWaterLevel(0.8f),
+evaporationRate(0.25f),
+fishLivingIn(0)
+{
+    std::cout << "a " << std::fixed << waterCapacity << "G AquariumTank was created" << std::endl; 
+}
```

Comment:
You can remove some duplication here if you create an `init` function.
inside that `init` function, set the value of the variables that are shared between these 2 ctor implementations.
Call `init` inside the constructor. 
the only variable that would be set from the 2nd constructor is the `waterCapacity`
```
AquariumTank::AquariumTank()
{
    init();
    std::cout << blah blah blah
}

AquariumTank::AquariumTank(float c) : waterCapacity(capacity)
{
    init();
    std::cout << blah blah blah
}
```
- [ ] revise